File: api/src/EventHub.Domain/ReadModels/ContributorDto.cs
namespace EventHub.Domain.ReadModels;

/// <summary>Contributeur affiché sur la page « À propos » (mobile + back office).</summary>
public sealed record ContributorDto
{
    public required Guid Id { get; init; }
    public required string Name { get; init; }
    public required string Role { get; init; }
    public string? AvatarUrl { get; init; }
    public int SortOrder { get; init; }
}

File: backoffice/src/app/features/settings/models/contributorDto.ts
/** Contributeur de la page « À propos » (GET/POST/PATCH /api/admin/contributors). */
export interface ContributorDto {
  id: string;
  name: string;
  role: string;
  avatarUrl?: string | null;
  sortOrder: number;
}

/** Payload create/update d'un contributeur. */
export interface ContributorRequest {
  name: string;
  role: string;
  avatarUrl?: string | null;
  sortOrder: number;
}
